### PR TITLE
SyliusPriceHistoryPlugin initialization

### PR DESCRIPTION
**Pull request by Rafikooo**

### CODE REVIEW

**Comment by Rafikooo** [comment]
Last build on the fork: https://github.com/Rafikooo/PriceHistoryPlugin/actions/runs/3950753920

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+backend:
+	tests/Application/bin/console sylius:install --no-interaction
+	tests/Application/bin/console sylius:fixtures:load default --no-interaction
+
+frontend:
+	(cd tests/Application && yarn install --pure-lockfile)
+	(cd tests/Application && GULP_ENV=prod yarn build)
+
+behat:
+	APP_ENV=test vendor/bin/behat --colors --strict --no-interaction -vvv -f progress
```
Comment (coldic3) [inline Makefile:28]: There is no difference between this line and line 14.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+services:
```
Comment (coldic3) [inline docker-compose.yml:1]: Let's add `docker-compose.override.yml` to the `.gitignore`.

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,10 @@
+parameters:
+    level: max
+    reportUnmatchedIgnoredErrors: false
```
Comment (coldic3) [inline phpstan.neon:3]:
```suggestion
    reportUnmatchedIgnoredErrors: true
```
And probably then we'll see the ignored error in line 10 should be removed.

File: tests/Application/config/packages/test/security.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+security:
+    password_hashers:
+        Sylius\Component\User\Model\UserInterface:
+            algorithm: argon2i
```
Comment (coldic3) [inline tests/Application/config/packages/test/security.yaml:4]: `plaintext` could be used instead

File: tests/Application/public/index.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+use Tests\Sylius\PriceHistoryPlugin\Application\Kernel;
+use Symfony\Component\ErrorHandler\Debug;
+use Symfony\Component\HttpFoundation\Request;
+
+require dirname(__DIR__) . '/config/bootstrap.php';
```
Comment (coldic3) [inline tests/Application/public/index.php:9]: We can use SF Runtime, eg. https://github.com/Sylius/Sylius-Standard/commit/2b99bdbf8cc6e715ac907ef7306120cc3cbdfabb

File: tests/Application/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+const path = require('path');
```
Comment (coldic3) [inline tests/Application/webpack.config.js:1]: I would be for this improvement https://github.com/Sylius/Sylius-Standard/commit/e97cb7c578ac071ab6cb4438b252be1dbd85abe5 just to be consistent.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+###> symfony/framework-bundle ###
+/.env.local
+/.env.local.php
+/.env.*.local
+/config/secrets/prod/prod.decrypt.private.php
+/public/bundles/
+/var/
+/vendor/
+###< symfony/framework-bundle ###
```
Comment (GSadee) [inline .gitignore:27]: We don't have some of these files and directories in the main directory, but in the `tests/Application`
```suggestion
###> symfony/framework-bundle ###
/config/secrets/prod/prod.decrypt.private.php
/public/bundles/
/vendor/
###< symfony/framework-bundle ###
```

File: phpunit.xml.dist [hunk 8]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/8.5/phpunit.xsd"
+         colors="true"
+         bootstrap="tests/Application/config/bootstrap.php">
+    <testsuites>
+        <testsuite name="SyliusPriceHistoryPlugin Test Suite">
```
Comment (GSadee) [inline phpunit.xml.dist:8]: It would be probably better to use snake case for the suite name

File: tests/Application/.env [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+APP_ENV=dev
+APP_DEBUG=1
+APP_SECRET=EDITME
+###< symfony/framework-bundle ###
+
+###> doctrine/doctrine-bundle ###
+# Format described at http://docs.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/configuration.html#connecting-using-a-url
+# For a sqlite database, use: "sqlite:///%kernel.project_dir%/var/data.db"
+# Set "serverVersion" to your server version to avoid edge-case exceptions and extra database calls
+DATABASE_URL=mysql://root@127.0.0.1/sylius_price_history_plugin_%kernel.environment%?serverVersion=5.7
```
Comment (GSadee) [inline tests/Application/.env:15]: IMO server version is not needed

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Rafał! :1st_place_medal:

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -0,0 +1,87 @@
+name: Build
```
[the remark this answers is not in the document]
Comment (Rafikooo) [reply .github/workflows/build.yml:1]: I'm not sure about this particular file. In each repository, we use the`yml` extension for the GitHub workflow

File: phpunit.xml.dist (hunk 8 above)
Comment (Rafikooo) [reply phpunit.xml.dist:8]: I was inspired by these example settings:

https://github.com/Sylius/Sylius/blob/1.13/phpunit.xml.dist#L11
https://github.com/Sylius/Plus/blob/main/phpunit.xml.dist#L8
https://github.com/Sylius/PayPalPlugin/blob/1.5/phpunit.xml.dist
